src/sql/workbench/services/executionPlan/common/executionPlanService.ts

```diff
@@ -27,19 +27,46 @@ export class ExecutionPlanService implements IExecutionPlanService {
 		this._providerRegisterEvent = this._onProviderRegister.event;
 	}
 
-	private async ensureCapabilitiesRegistered(): Promise<void> {
-		let providers = Object.keys(this._capabilitiesService.providers);
-		if (!providers) {
-			await new Promise<void>(resolve => {
-				this._capabilitiesService.onCapabilitiesRegistered(e => {
-					resolve();
+	/**
+	 * This ensures that the capabilities service has registered the providers to handle execution plan requests.
+	 */
+	private async ensureCapabilitiesRegistered(providerId?: string): Promise<void> {
+		if (!providerId) {
```

Comment:
So this case still confuses me - are we just going to blindly pick the first provider that's registered and hope that it can handle it? Shouldn't we still be looking for a provider that's able to handle the file with the extension of the plan?